### DISCUSSION TITLE
When analyzing data using binary data, other values ​​are additionally created in the middle.

### DISCUSSION THREAD

**Discussion by boozesounds** [ourownstory/neural_prophet#1158] [Q&A - get help using NeuralProphet] [answered]
```
m = NeuralProphet(
    n_forecasts=3,
    n_lags=3,
    learning_rate=0.1,
    normalize=False
)
m = m.highlight_nth_step_ahead_of_each_forecast(3)
metrics = m.fit(df, freq="D")
```

![binarydata](https://user-images.githubusercontent.com/35713030/212606076-4669722d-a9b7-4c7f-9e29-fe57e08073b6.png)

I only put 2 and 3 data. Likewise When you put binary data like 0 and 1, 1 and -1 another value is added to the middle line of the data to start the analysis. How can I fix this situation that occurs when I put in binary time series data?

and this problem occurs when using lag .

**Comment by leoniewgnr** [comment]
Hi @boozesounds, thanks for your question! Do you have a dataset that you can share which leads to that behaviour or is it only data consisting of 2 and 3?

**Comment by judussoari** [comment, marked as answer]
@boozesounds I assume the values are imputed. Try setting `impute_missing=False` when defining `m = NeuralProphet()`.
